Sources/Sentry/SentryBreadcrumbTracker.m

```diff
@@ -129,17 +132,23 @@ - (void)trackApplicationUIKitNotifications
 #if !TARGET_OS_WATCH
 - (void)trackNetworkConnectivityChanges
 {
-    [SentryDependencyContainer.sharedInstance.reachability
-         addObserver:self
-        withCallback:^(BOOL connected, NSString *_Nonnull typeDescription) {
-            SentryBreadcrumb *crumb =
-                [[SentryBreadcrumb alloc] initWithLevel:kSentryLevelInfo
-                                               category:@"device.connectivity"];
-            crumb.type = @"connectivity";
-            crumb.data = [NSDictionary dictionaryWithObject:typeDescription forKey:@"connectivity"];
-            [self.delegate addBreadcrumb:crumb];
```

Comment:
The `self.delegate` is already weak reference see https://github.com/getsentry/sentry-cocoa/blob/3b38d9a8191862eacf58d85c7f550ac913ce5164/Sources/Sentry/SentryBreadcrumbTracker.m#L32 so we don't need to make it a weak self. 

I personally prefer the delegate pattern, cause you don't need to worry about using weak references in the block if you set it up properly. Otherwise, you have to ensure using weak references in the blocks which is easier to get wrong.

Reply:
> couldn't we use the NSDictionary with the hashes as keys instead of the problematic descriptions

Yes, if we use `NSMapTable` as pointed out by you, but I would prefer the `NSHashTable`, cause we don't have to worry about how to come up with the keys and write `.allValues` two times. I can change it if you prefer. No strong opinion on this one.

Reply:
Nope, I'm fine with all this!